addons/website/static/src/js/new_content_form.js

```diff
@@ -1,24 +1,37 @@
 /** @odoo-module **/
 
-import FormController from 'web.FormController';
+import {formView} from "@web/views/form/form_view";
+import {useService} from "@web/core/utils/hooks";
+import {registry} from "@web/core/registry";
 
-export const NewContentFormController = FormController.extend({
-    /**
-     * @override
-     */
-    saveRecord() {
-        return this._super.apply(this, arguments).then(() => {
-            const state = this.model.get(this.handle);
-            this.do_action({
+export class NewContentFormController extends formView.Controller {
+    setup() {
+        super.setup();
+        this.action = useService('action');
+    }
+
+    async save() {
+        await super.save();
+        if (this.model.root.resId) {
+            this.action.doAction({
                 type: 'ir.actions.act_window_close',
-                infos: {path: this._getPath(state)},
+                infos: {path: this.path},
             });
-        });
-    },
-    /**
-     * @private
-     */
-    _getPath(state) {
-        return state.data.website_url;
+        }
+    }
+
+    get path() {
+        return this.model.root.data.website_url;
```

Comment:
There were documentation comment markers before, you removed them. Should not you keep them? Actually a bit of explanation on the purpose of "path" compared to a new normal form view controller could be nice